xbmc/addons/AddonManager.cpp

```diff
@@ -830,6 +831,7 @@ bool CAddonMgr::DisableAddon(const std::string& id)
   AddonPtr addon;
   if (GetAddon(id, addon, ADDON_UNKNOWN, false) && addon != NULL)
   {
+    CServiceBroker::GetBinaryAddonManager().DisableEvent(addon->Type(), addon->ID());
```

Comment:
Not really sure why GetBinaryAddonManager()->*Event is called from here. Would make more sense to just Post/Send the events instead of going through an extra method to do it.